src/python/pants/backend/go/pkg.py

```diff
@@ -248,15 +282,68 @@ async def resolve_go_package(
     )
 
 
+@rule
+async def resolve_external_go_package(
+    request: ResolveExternalGoPackageRequest,
+) -> ResolvedGoPackage:
+    wrapped_target = await Get(WrappedTarget, Address, request.address)
+    target = wrapped_target.target
+
+    import_path = target[GoImportPath].value
+    if not import_path:
+        raise ValueError(
+            f"_go_ext_mod_package at address {request.address} does not have an import path set."
+        )
```

Comment:
This should be expressed via the Target API. Set `required = True`. It's awkward that you need to override the type hint for `value` so MyPy understands it's required. In the class, set `value: str` for example, whereas it defaults to `Optional[str]`.